File: src/main/java/com/baulsupp/oksocial/credentials/CredentialFactory.java
package com.baulsupp.oksocial.credentials;

import com.mcdermottroe.apple.OSXKeychainException;
import ee.schimke.oksocial.output.util.PlatformUtil;
import java.util.Optional;

public class CredentialFactory {
  public static CredentialsStore createCredentialsStore(Optional<String> tokenSet)
      throws OSXKeychainException {
    if (PlatformUtil.isOSX()) {
      return new OSXCredentialsStore(tokenSet);
    } else {
      return new PreferencesCredentialsStore(tokenSet);
    }
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/credentials/CredentialFactory.kt
package com.baulsupp.oksocial.credentials

import com.baulsupp.oksocial.output.util.PlatformUtil

object CredentialFactory {
  fun createCredentialsStore(): CredentialsStore {
    return if (PlatformUtil.isOSX) {
      OSXCredentialsStore()
    } else {
      PreferencesCredentialsStore()
    }
  }
}
